lms/djangoapps/course_api/views.py

```diff
@@ -206,3 +213,43 @@ def get_queryset(self):
             org=form.cleaned_data['org'],
             filter_=form.cleaned_data['filter_'],
         )
+
+
+@view_auth_classes(is_authenticated=False)
+class CourseDeletionView(DeveloperErrorViewMixin, APIView):
+    """
+    **Use Cases**
+
+        Request details for a course
+
+    **Example Requests**
+
+        GET /api/courses/v1/courses/delete_course/{course_key}/
+
+    **Parameters:**
+
+        course_key:
+            The course key for deletion.
+
+    **Returns**
+
+        * 200 on success with above fields.
+        * 400 if the course is not available or cannot be seen.
+    """
+
+    serializer_class = CourseDetailSerializer
+
+    def get(self, request, *args, **kwargs):
+
+        course_key_string = self.kwargs['course_key_string']
+        try:
+            course_key = CourseKey.from_string(course_key_string)
+        except InvalidKeyError:
+            raise ValidationError("Invalid course_key: '{}'.".format(course_key_string))
+
+        if not modulestore().get_course(course_key):
+            raise ValidationError("Course with '{}' key not found.".format(course_key_string))
```

Comment:
I am not sure this should return `400`. RFC for delete says it should be 204 in case of no entity returned.